### ISSUE TITLE
ClickHouse: SELECT * EXCEPT col without parentheses is unparsable

### ISSUE THREAD

**Issue by friedrich123456789** [sqlfluff/sqlfluff#8034]
### What happened

The parenthesised column-exclusion modifier `* EXCEPT (col)` parses, but the
unparenthesised single-column form `* EXCEPT col` does not (also `t.* EXCEPT col`).

Related: #7654 added `APPLY(...)` and #7999 fixed an RF02 false positive on
`* EXCEPT (...)`. The remaining gap is the unparenthesised `EXCEPT` spelling.

### Expected behaviour

Both `* EXCEPT col` and `* EXCEPT (c1, c2)` parse.

### Observed behaviour

```
SELECT * EXCEPT c FROM t    ->  L1:P10 PRS: Found unparsable section: 'EXCEPT c FROM t'
SELECT t.* EXCEPT c FROM t  ->  L1:P12 PRS: Found unparsable section: 'EXCEPT c FROM t'
```

For contrast, `SELECT * EXCEPT (c) FROM t` and `SELECT * EXCEPT (c1, c2) FROM t`
parse fine.

### How to reproduce

```sql
SELECT * EXCEPT c FROM t
SELECT t.* EXCEPT c FROM t
```

```bash
sqlfluff parse --dialect clickhouse -
```

Valid, verified on ClickHouse 26.4.2.
Docs: https://clickhouse.com/docs/sql-reference/statements/select#except

### Dialect

clickhouse

### Version

sqlfluff 4.2.2 (latest release), Python 3.12; also applies to `main`.

### Configuration

None — default rules, `--dialect clickhouse`.

**Comment by peterbud** [comment]
Hi @friedrich123456789 thanks for opening this PR.

I have checked again the `EXCEPT` modifier syntax here: https://clickhouse.com/docs/sql-reference/statements/select/except-modifier, but it seems that the parenthesis IS mandatory. I understand that the engine might parse the expression without parentheses, but based on what I see it's syntactically not matching the specs.

Wat do you think?

**Comment by friedrich123456789** [comment]
> Hi [@friedrich123456789](https://github.com/friedrich123456789) thanks for opening this PR.
> 
> I have checked again the `EXCEPT` modifier syntax here: https://clickhouse.com/docs/sql-reference/statements/select/except-modifier, but it seems that the parenthesis IS mandatory. I understand that the engine might parse the expression without parentheses, but based on what I see it's syntactically not matching the specs.
> 
> Wat do you think?

Hi @peterbud, 
I agree, the clickhouse docs do not show the (in practice valid) single column no parenthesis case. I have requested an [update to the clickhouse docs](https://github.com/ClickHouse/ClickHouse/pull/109293). Let's see what they say. If they deny my change request to the docs you can close this issue.

**Comment by peterbud** [comment]
Hi @friedrich123456789 , I have checked the issue you have opened: that repo has 4800+ open issues, so I'd suggest to close this issue here, and re-open it once you hear back in a positive way from the Clickhouse team.
Thanks!

**Comment by friedrich123456789** [comment]
> Hi [@friedrich123456789](https://github.com/friedrich123456789) , I have checked the issue you have opened: that repo has 4800+ open issues, so I'd suggest to close this issue here, and re-open it once you hear back in a positive way from the Clickhouse team. Thanks!

sure, thank you
